src/http.ts

```diff
@@ -30,6 +39,131 @@ export interface SourceImageOptions {
   requestEtag?: string
 }
 
+interface UsageTrackingItem {
+  runningCount: number;
+  cacheKey: string;
+  lastAccess: number;
+  inputCacheFile: string
+  size: number
+}
+
+type UsageTracking = Record<
+  string,
+  UsageTrackingItem
+>;
+
+const USAGE_TRACKING_KEY = 'usage-tracking'
+const trackingLock = new Lock()
+
+async function getTracking (metadataStore: Storage): Promise<UsageTracking> {
+  return ((await metadataStore.getItem(USAGE_TRACKING_KEY)) as
+      | UsageTracking
+      | undefined) ?? {}
+}
+
+async function markUsageStart (
+  metadataStore: Storage,
+  cacheKey: string,
+  inputCacheFile: string
+): Promise<void> {
+  await trackingLock.acquire()
+  try {
+    const tracking = await getTracking(metadataStore)
+
+    let usageTrackingItem = tracking[cacheKey]
+    if (!usageTrackingItem) {
+      tracking[cacheKey] = usageTrackingItem = {
+        runningCount: 0,
+        lastAccess: 0,
+        size: 0,
+        cacheKey,
+        inputCacheFile
+      }
+    }
+
+    usageTrackingItem.runningCount++
+    usageTrackingItem.lastAccess = Date.now()
+
+    await metadataStore.setItem(USAGE_TRACKING_KEY, tracking)
+  } finally {
+    trackingLock.release()
+  }
+}
+
+async function markUsageComplete (
+  metadataStore: Storage,
+  cacheKey: string
+
+) {
+  await trackingLock.acquire()
+  try {
+    const tracking = await getTracking(metadataStore)
+
+    const usageTrackingItem = tracking[cacheKey]
+    if (usageTrackingItem) {
+      usageTrackingItem.runningCount--
+
+      if (await pathExists(usageTrackingItem.inputCacheFile)) {
+        const { size } = await stat(usageTrackingItem.inputCacheFile)
+        usageTrackingItem.size = size
+      } else {
+        // If the file doesn't exist, we can't track it
+        delete tracking[cacheKey]
```

Comment:
what's the scenario where the file doesn't exist? Is it being potentially already evicted from the tracking cache the scenario?

Reply:
Entry is added when first request for given source image url starts. At this point we don't yet know wether downloading source image will succeed (url might be wrong and it will never work, url might be temporarily unavailable, actual downloading might fail for whatever network reason etc). In this cases there is no size available because file doesn't exist so might as well cleanup that entry